### DISCUSSION TITLE
AMP support for Next.js 13 project with server components

### DISCUSSION THREAD

**Discussion by SettarovAli** [vercel/next.js#48243] [Help]
### Summary

I am migrating project from Next 12 to Next 13 and facing problem with AMP support.

As I can see from the Next 13 docs AMP support will not be released - https://beta.nextjs.org/docs/app-directory-roadmap#not-planned-features

How to implement AMP support for Next.js 13 project with server components?

**Comment by PegeDev** [comment]
any update for this problem?

**Comment by kozhuhds** [comment]
good question! even if it was supposed AMP to be implemented manually, currently there is no way to add `<link rel="amphtml" href="https://www.example.com/url/to/amp/document.html">` to a non-amp page with the new `generateMetadata`

**Comment by Andrii-Moshenskyi** [comment]
I have the same question. Any updates on this?

Привіт @SettarovAli ! Was you succeed to migrate AMP pages to Next.js 13?

**Comment by kozhuhds** [comment]
wrote the blog posts with workarounds for **static export** (both problems: inserting amphtml link tag and support for amp in next.js 13)
[https://www.kozhuhds.com/blog/next-13-static-export-for-production#amp-support](https://www.kozhuhds.com/blog/next-13-static-export-for-production#amp-support)
[https://www.kozhuhds.com/blog/how-to-build-a-static-mdx-blog-with-nextjs-and-contentlayer#step-6-accelerated-mobile-pages-amp](https://www.kozhuhds.com/blog/how-to-build-a-static-mdx-blog-with-nextjs-and-contentlayer#step-6-accelerated-mobile-pages-amp)

**Reply by cellva01** [reply]
can't you just add that link by js dynamically like inserting a new dom element in head tag?

**Reply by kozhuhds** [reply]
@cellva01 not sure if google crawlers can read dynamically added elements

**Reply by cellva03** [reply]
Yoo brother you could also use the nextjs 13 newly introduced generate metatag function to generate link the amphtml to the dom .

export const metadata = {
  icons: {
    other: {
      rel: 'amphtml',
      url: 'url here',
    },
  },
}

**Comment by KamilOcean** [comment]
Hey, guys! I've found a crutch solution to add the `<link rel="amphtml" href="..." />` tag into your header via `icon` config.

```ts
export async function generateMetadata({ params }): Promise<Metadata> {
  return {
    icons: {
      other: {
        rel: "amphtml",
        url: `${YOUR_SITE_HOST}/amp/${params.slug.join('/')}`
      }
    }
  }
}
```

The resulting template:

```html
<link rel="amphtml" href="https://example.com/amp/articles/bla/bla/bla/23">
```

I've written a [gist](https://gist.github.com/KamilOcean/0692be9168c7f672c7ac250d7b90dfb1) about this and linked the solution of @kozhuhds.

*UPD*
@cellva03 I'm sorry, I've done a big research and written the `gist` and only then I realized that you already solved it. I just thought in your unformatted code, that you create `<meta name="amphtml"...` tag.
